### PR TITLE
feat(solc): Strip experimental pragma from all imported contracts

### LINKED ISSUE #1115: [ethers-solc] `pragma experimental` is not stripped on flatten
**Version**
List the versions of all `ethers-rs` crates you are using. The easiest way to get
this information is using `cargo-tree`.

`cargo install cargo-tree`
(see install here: https://github.com/sfackler/cargo-tree)

Then:

`cargo tree | grep ethers`

```
✦ forge --version
forge 0.2.0 (233ab70 2022-04-01T00:48:24.366564+00:00)
```

I think foundry uses `ethers-rs` on master branch.

**Platform**
The output of `uname -a` (UNIX), or version and 32 or 64-bit (Windows)

```
✦ uname -a
Darwin pyk-iMac.local 21.4.0 Darwin Kernel Version 21.4.0: Mon Feb 21 20:36:53 PST 2022; root:xnu-8020.101.4~2/RELEASE_ARM64_T8101 x86_64
```

**Description**
Enter your issue details here.
One way to structure the description:

`pragma experimental` is not stripped on flatten

I tried this code:

```
forge flatten src/Contract.sol > output.sol
```

I expected to see this happen:

`pragma experimental ABIEncoderV2;` will be stripped on each import.

Instead, this happened: 

`pragma experimental ABIEncoderV2;` is not stripped, this issue cause the verify contract on Forge to failed:

```
Error:
   0: Contract verification failed:
      Response: `NOTOK`
      Details: `Fail - Unable to verify`
```

--

https://github.com/gakonst/foundry/issues/852

### PR DESCRIPTION
**Pull request by pyk**
## Motivation

`forge verify-contract` failed because `flatten` doesn't strip the experimental pragma from imported contracts.

Related issues: 
https://github.com/gakonst/foundry/issues/852
https://github.com/gakonst/ethers-rs/issues/1115

## Solution

This PR enable `solc.flatten` to strip pragma from all imported contracts.

Fix #1115 

## PR Checklist

- [x] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

**Comment by pyk** [comment]
@mattsse @gakonst is there any simple way to test this out directly on foundry?

AFAIK `foundryup` only support installing `forge` using specific github repo, cannot specify which `ethers-rs` is used.

**Comment by pyk** [comment]
nvm guys, I just learned that I can patch the `ethers-rs` using my local repos. https://github.com/gakonst/foundry/blob/6130af7d3fd581c22b42d232011fa1576bfc6ac0/Cargo.toml#L25-L32

btw here is the example contract successfully verified after stripping the experimental pragma

https://arbiscan.io/address/0x1B82DF960F5470a17606abAd2E541CC552870098#code

Previously it always failed to verify coz `flatten` doesn't strip the experimental pragma on imported files

**Review by gakonst — APPROVED** [review]
This LGTM, defer to @mattsse approval before merging. Good catch, thank you!

**Review by mattsse — APPROVED** [review]
awesome! lgtm

**Comment by pyk** [comment]
@mattsse ty ser!

**Pull request merged**
